staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -584,6 +657,19 @@ type sharedProcessor struct {
 	wg               wait.Group
 }
 
+func (p *sharedProcessor) isRegistered(listener *processorListener) bool {
+	p.listenersLock.Lock()
+	defer p.listenersLock.Unlock()
+
+	for i := 0; i < len(p.listeners); i++ {
```

Comment:
Is there any reason not to switch the lists to maps/sets? e.g. make the type `map[*processorListener]struct{}`. Then this check and the removal are easy.